src/camlsnark_c/libsnark-caml/libsnark/caml/common.cpp

```diff
@@ -7,79 +7,124 @@
 #include <libff/algebra/curves/mnt/mnt6/mnt6_pp.hpp>
 #include <libff/common/profiling.hpp>
 #include <stdarg.h>
-
-int (*snarky_printf_deferred)(const char* format, va_list args) = &vprintf;
-
-int snarky_printf(const char* format, ...)
-{
-    va_list args;
-    va_start(args, format);
-    int ret = (*snarky_printf_deferred)(format, args);
-    va_end(args);
+#include <time.h>
+
+int normal_printf(const char* filename,
+                  int line,
+                  FILE* file,
+                  const char* format,
+                  va_list args) {
+    return vfprintf(file, format, args);
+}
+
+int json_printf(const char* filename,
+                int line,
+                FILE* file,
+                const char* format,
+                va_list args) {
+    time_t now;
+    struct tm* now_utc;
+
+    time(&now);
+    now_utc = gmtime(&now);
+    fprintf(file,
+            "{\"timestamp\":\"%d-%02d-%02d %02d-%02d:%02d:%06dZ",
+            now_utc->tm_year + 1900,
+            now_utc->tm_mon,
+            now_utc->tm_mday,
+            now_utc->tm_hour,
+            now_utc->tm_min,
+            now_utc->tm_sec,
+            0);
+    fprintf (file, "\",\"level\":\"%s\"", "DEBUG");
+    fprintf (file,
+             ",\"source\":{\"module\":\"%s\",\"location\":\"File \"%s\", line %d, characters 0-1\"},\"message\":\"",
```

Comment:
I think this results in invalid JSON? The inner `"` after `File` need to be escaped.